streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -177,9 +177,12 @@ class JobScheduler(val ssc: StreamingContext) extends Logging {
             jobSet.totalDelay / 1000.0, jobSet.time.toString,
             jobSet.processingDelay / 1000.0
           ))
-          listenerBus.post(StreamingListenerBatchCompleted(jobSet.toBatchInfo))
+          listenerBus.post(StreamingListenerBatchCompleted(jobSet.toBatchInfo()))
         }
       case Failure(e) =>
+        val jobSet = jobSets.get(job.time)
```

Comment:
Should this line be moved above the `job.result match {` because both cases seem to use it.